packages/@react-spectrum/provider/test/Provider.test.js

```diff
@@ -283,4 +283,47 @@ describe('Provider', () => {
       expect(onRender).toHaveBeenCalledTimes(2);
     });
   });
+
+  describe('responsive breakpoints', function () {
+    const testWidths = {base: '100px', S: '200px', M: '300px', L: '400px'};
+    const testBreakpoints = {
+      [testWidths.L]: {S: 480, M: 640, L: 1024},
+      [testWidths.M]: {S: 640, M: 1024, L: 1168},
+      [testWidths.S]: {S: 1024, M: 1168, L: 1440},
+      [testWidths.base]: {S: 1168, M: 1440, L: 1680}
+    };
+
+    // render once to get a reference to the rerender method
+    let {container, rerender} = render(
+      <Provider breakpoints={testBreakpoints[testWidths.L]} theme={theme}>
+        <TextField
+          label="foo"
+          width={testWidths} />
+      </Provider>
+    );
+    let provider, textField;
+
+    // only rerender, updating the breakpoints prop (all others constant)
+    it.each`
+      name                                 | breakpoints                         | expected
+      ${'(L) ' + testWidths.L + ':'}       | ${testBreakpoints[testWidths.L]}    | ${testWidths.L}
+      ${'(M) ' + testWidths.M + ':'}       | ${testBreakpoints[testWidths.M]}    | ${testWidths.M}
+      ${'(S) ' + testWidths.S + ':'}       | ${testBreakpoints[testWidths.S]}    | ${testWidths.S}
+      ${'(base) ' + testWidths.base + ':'} | ${testBreakpoints[testWidths.base]} | ${testWidths.base}
+    `('$name $breakpoints', function ({breakpoints, expected}) {
+      matchMedia.useMediaQuery('(min-width: 1024px)');
+
+      rerender(
+        <Provider breakpoints={breakpoints} theme={theme}>
+          <TextField
+            label="foo"
+            width={testWidths} />
+        </Provider>
+      );
+
+      provider = container.firstChild;
+      textField = provider.firstChild;
+      expect(textField).toHaveStyle({width: expected});
```

Comment:
```suggestion
      let textfield = getByRole('textbox');
      expect(textField).toHaveStyle({width: expected});
```

Reply:
Hmmm... so, I _think_ I originally tried something like this, but the generated HTML is something like:

```
<div
  class="spectrum-Field spectrum-Field--positionTop"
  style="width: 100px;"
>
  // sibling elements omitted for brevity
  <div
    class="flex spectrum-Field-wrapper"
    style="flex-direction: column;"
  >
    <div
      class="spectrum-Textfield spectrum-Field-field"
    >
      <input
        aria-labelledby="react-aria1656127020-14"
        class="spectrum-Textfield-input"
        id="react-aria1656127020-13"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
```

...and, the `getByRole` method only matches this element:

```
<input
  aria-labelledby="react-aria1656127020-14"
  class="spectrum-Textfield-input"
  id="react-aria1656127020-13"
  type="text"
  value=""
/>
```
...so, it's difficult to get at the styles applied at different breakpoints by drilling that far into the DOM.

Any other ideas here (e.g. - element reference and getBoundingClientRect, etc)?

Reply:
...of course, methods to get at a rendered element's actual dimensions (positioning, etc) aren't going to work with jsdom (headless environment).

It may be easier to continue checking against the component container here, since the appropriate styles are being set there in these scenarios

Reply:
I checked again and setting a test id on the text field itself (and then retrieving a reference by test id later) results in the inner input element being matched, too (instead of the outer div that contains this element - basically, the same limitation as fetching by role)

Reply:
Found a nice solution that avoids limitations with using the container and DOM structure